src/app/views/datasets-new/create/CreateCantabularDatasetController.jsx

```diff
@@ -24,24 +25,57 @@ export class CreateCantabularDatasetController extends Component {
 
         this.state = {
             isPosting: false,
+            isGettingRecipe: false,
             datasetID: "",
             format: "",
+            isBasedOn: "",
         };
     }
 
     UNSAFE_componentWillMount = () => {
         this.setStateFromParameters();
+        this.getRecipe();
     };
 
     setStateFromParameters = () => {
         this.setState({
             datasetID: this.props.params.datasetID,
-            format: this.props.params.format,
         });
     };
 
+    handleGETSuccess = recipe => {
+        this.setState({
+            format: recipe.format,
+            isBasedOn: recipe.cantabular_blob,
+        });
+    };
+
+    getRecipe = () => {
+        const recipeId = this.props.params.recipeID;
```

Comment:
Just a question, can `this.props.params` be `undefined` - if so, `this.props.params.recipeID` might throw a reference error....